Utils/Dataflow/092_datasetInfoAMI/amiDatasets.py

```diff
@@ -0,0 +1,123 @@
+#!/bin/env python
+import json
+import re
+import sys
+import os
+try:
+    import pyAMI.client
+    import pyAMI.atlas.api as AtlasAPI
+    import pyAMI.config
+except ImportError:
+    sys.stderr.write("Unable to find pyAMI client.")
+    sys.exit(1)
+
+base_dir = os.path.abspath(os.path.dirname(__file__))
+
+try:
+    dkb_dir = os.path.join(base_dir, os.pardir)
+    sys.path.append(dkb_dir)
+    import pyDKB
+except Exception, err:
+    sys.stderr.write("(ERROR) Failed to import pyDKB library: %s\n" % err)
+    sys.exit(1)
+
+ami_client = None
+PHYS_VALUES = [{"ami": "genFiltEff", "es": "gen_filt_eff"},
+              {"ami": "crossSection", "es": "cross_section"},
+              {"ami": "crossSectionRef", "es": "cross_section_ref"},
+              {"ami": "kFactor", "es": "k_factor"}]
+
+def main(argv):
+
+    stage = pyDKB.dataflow.stage.JSONProcessorStage()
+
+    stage.add_argument('--userkey', help='PEM key file', required=True)
+    stage.add_argument('--usercert', help='PEM certificate file', required=True)
+
+    exit_code = 0
+    try:
+        stage.parse_args(argv)
+        stage.process = process
+        init_ami_client(stage.ARGS.userkey, stage.ARGS.usercert)
+        stage.run()
+    except (pyDKB.dataflow.exceptions.DataflowException, RuntimeError), err:
+        if str(err):
+            str_err = str(err).replace("\n", "\n(==) ")
+            sys.stderr.write("(ERROR) %s\n" % str_err)
+        exit_code = 2
+    finally:
+        stage.stop()
+
+    sys.exit(exit_code)
+
+def init_ami_client(userkey, usercert):
+    """
+    Initialisation of AMI client into the global variable
+    :param userkey: user key pem file
+    :param usercert: user certificate pem file
+    :return:
+    """
+    global ami_client
+    try:
+        ami_client = pyAMI.client.Client('atlas', key_file=userkey, cert_file=usercert)
+        AtlasAPI.init()
+    except:
+        sys.stderr.write(
+            "Could not establish pyAMI session. Are you sure you have a valid certificate?")
+        sys.exit(1)
+
+def process(stage, message):
+
+    str = amiPhysValues(message.content())
+    stage.output(pyDKB.dataflow.messages.JSONMessage(amiPhysValues(message.content())))
+
```

Comment:
Two comments in a row:
* `str` is a Python keyword (namely the type), so reusing it as a variable name you can invite unpredictable behaviour in the code (say, in conditions like `if type(x) is str`);
* `amiPhysValues()` is called twice